### ISSUE TITLE
Change TextBlock.Text in invisible StackPanel with horizontal orientation

### ISSUE THREAD

**Issue by RomanTimv** [AvaloniaUI/Avalonia#17492]
### Describe the bug

TextBlock doesn't display text if it was changed when containing StackPanel with horizontal orientation was invisible.
Grid or vertical StackPanel works correctly

### To Reproduce

Place on View:
- StackPanel Orientation="Vertical" HorizontalAlignment="Stretch" Background="LightGreen" IsVisible="False"
- TextBlock inside StackPanel
- Button

In Button.Click handler set TextBox.Text to "Test" and StackPanel.IsVisible to true

StackPanel becomes visible (green stripe) but without text in it

### Expected behavior

StackPanel text is visible in StackPanel

### Avalonia version

11.2.0

### OS

Windows

**Comment by Gillibald** [comment]
Please test against current master

**Comment by RomanTimv** [comment]
Same behavior.

```
  <Grid>
    <Grid.RowDefinitions>
      <RowDefinition Height="Auto" />
      <RowDefinition Height="Auto" />
      <RowDefinition Height="Auto" />
    </Grid.RowDefinitions>

    <StackPanel  Name="cv" Orientation="Vertical" Grid.Row="0" Margin="4" HorizontalAlignment="Stretch" Background="LightGreen" IsVisible="False">
      <TextBlock Name="tv" />
    </StackPanel>

    <StackPanel Name="ch" Orientation="Horizontal" Grid.Row="1" Margin="4" HorizontalAlignment="Stretch" Background="LightGreen" IsVisible="False">
      <TextBlock Name="th" />
    </StackPanel>

    <Button Grid.Row="2" Margin="4" Content="Show" Click="Button_Click" />
  </Grid>
```
```
        private void Button_Click(object sender, RoutedEventArgs e)
        {
            cv.IsVisible = true;
            tv.Text = "Test";

            ch.IsVisible = true;
            th.Text = "Test";
        }
```
On button click I got
![image](https://github.com/user-attachments/assets/e5568bfa-4055-4116-a152-59a40e035063)

**Comment by Gillibald** [comment]
Should be fixed by: #17638
